src/graphics/ui/Chat.ts

```diff
@@ -0,0 +1,76 @@
+import * as Player from '../../entity/Player';
+import { Overworld } from '../../Overworld';
+import { hexToString, lightenColor } from './colorUtil';
+import { playerNoColor } from './colors';
+import { MESSAGE_TYPES } from '../../types/MessageTypes';
+
+export const elChatbox = document.getElementById('chatbox');
+export const elChatinput = <HTMLInputElement>document.getElementById('chatinput');
+export const elChatinner = document.getElementById('messages');
+var chatTimeout: NodeJS.Timeout;
+var NotificationTime = 5000;
+
+// send message when enter is pressed TODO: send message packet so players in the server can receive the message
+export function sendChatHandler(overworld: Overworld, e: KeyboardEvent) {
+  if (e.key == 'Enter') {
+    if (elChatinput.value.trim().length) {
+      // check for chat input after trimming whitespace
+      var chatname = 'Unknown'; // name of chatter
+      var chattercolor = hexToString(playerNoColor);
+      var message; // input
+      // var timestamp;  TODO: Include time?
+      if (globalThis.player?.name) {
+        chatname = globalThis.player.name;
+        chattercolor = globalThis.player?.color
+          ? hexToString(lightenColor(globalThis.player.color, 0.3))
+          : chattercolor;
+      }
+      message = elChatinput.value;
+      if (elChatinner) {
+        elChatinner.innerHTML += `<div class="message-speaker" style="color: ${chattercolor};">${chatname}:
+            <span class="text" style="color: white;">${message}</span></div> <br>`;
+      }
```

Comment:
Rather than having this logic in both sendChatHandler and RecieveMessage just have it in Recieve message and then don't exclude the current player from processing their own message.  This way you only have to maintain one area where chat text gets added to the chatbox

Reply:
This is the intent behind WebsocketPie, you can process messages you send and messages you receive in the same place :)

Reply:
Rewrote the section so all the logic is in receive message, and all we're sending is the message now